dataikuapi/govern/artifact_search_handler.py

```diff
@@ -0,0 +1,165 @@
+            ids_list = []
+
+        if artifact_search_source_type not in ["all", "blueprints", "blueprintVersions", "artifacts"]:
+            raise ValueError(""""Parameter artifact_search_source must be chosen from "all", "blueprints", 
+            "blueprintVersions", or "artifacts".""")
+
+        artifact_search_source = {"type": artifact_search_source_type}
+        if artifact_search_source_type == "blueprints":
+            artifact_search_source["blueprintIds"] = ids_list
+        if artifact_search_source_type == "blueprintVersions":
+            artifact_search_source["blueprintVersionIds"] = ids_list
+        if artifact_search_source_type == "artifacts":
+            artifact_search_source["artifactIds"] = ids_list
+
+        if artifact_filters_list is None:
+            artifact_filters_list = []
+        query = {"artifactFilters": [artifact_filter.build() for artifact_filter in artifact_filters_list]}
+        if sort_direction is not None or sort_column is not None:
+            query["artifactSearchSort"] = {"direction": sort_direction, "column": {"type": sort_column}}
+
+        artifact_search_pagination = {}
+        if page_size != 20:
+            artifact_search_pagination["pageSize"] = page_size
+        if last_artifact_id is not None:
+            artifact_search_pagination["lastArtifactId"] = last_artifact_id
+
+        return GovernArtifactSearchRequest(self.client, artifact_search_source, query, artifact_search_pagination)
+
+
+class GovernArtifactSearchRequest(object):
+    """
+    A search query object.
+    Do not create this directly, use :meth:`dataikuapi.govern.artifact_search_handler.build_query()` and then run the
+    query using :meth:`dataikuapi.govern.artifact_search_handler.GovernArtifactSearchRequest.perform_search()`
+    """
+
+    def __init__(self, client, artifact_search_source, query, artifact_search_pagination):
+        self.client = client
+        self.artifact_search_source = artifact_search_source
+        self.query = query
+        self.artifact_search_pagination = artifact_search_pagination
+
+    def perform_search(self):
+        """
+        Run the search request.
+
+        :returns The result of the search request.
+        :rtype: python dict. This dict contains a key "hasNextPage" and another one "uiArtifacts" which is the list of
+        artifacts.
+        """
+        return self.client._perform_json("POST", "/artifacts/search",
+                                         body=self.__build())
+
+    def __build(self):
+        return {"artifactSearchSource": self.artifact_search_source,
+                "query": self.query,
+                "artifactSearchPagination": self.artifact_search_pagination}
+
+
+class GovernFieldValueArtifactFilter(object):
+    """
+    A handle to apply filters on specific fields on a search request.
+    Do not create this directly, use :meth:`dataikuapi.govern.artifact_search_handler.build_field_artifact_filter()`
```

Comment:
```suggestion
    Do not create this directly, use :meth:`dataikuapi.govern.artifact_search_handler.ArtifactSearchHandler.build_field_artifact_filter()`
```

is more correct no ?